extensions/wikia/Flags/specials/scripts/FlagEditForm.js

```diff
@@ -0,0 +1,273 @@
+define('ext.wikia.Flags.FlagEditForm',
+	['jquery', 'mw', 'wikia.loader', 'wikia.cache', 'wikia.nirvana', 'wikia.mustache', 'BannerNotification'],
+	function ($, mw, loader, cache, nirvana, mustache, BannerNotification) {
+		/* In order of appearance */
+		var formData,
+			modalConfig,
+			resourcesCacheKey = 'flagEditFormResources',
+			emptyFormCacheKey = 'flagEditFormEmpty',
+			cacheVersion = '1.0';
+
+		/* Modal component configuration */
+		function init(prefillData) {
+			/**
+			 * FOR DEVELOPMENT ONLY
+			 */
+			cache.del(getResourcesCacheKey());
+			cache.del(getEmptyFormCacheKey());
+
+			$.when(getFormResources()).done(function (formResources) {
+				setupForm(formResources);
+
+				if (prefillData === undefined || prefillData.values === undefined) {
+					displayFormCreate();
+				} else {
+					displayFormEdit(prefillData);
+				}
+			});
+		}
+
+		function getFormResources() {
+			var formResources = cache.get(getResourcesCacheKey());
+
+			if (formResources == null || formResources.resources == null) {
+				formResources = loader({
+					type: loader.MULTI,
+					resources: {
+						messages: 'FlagsCreateForm',
+						mustache: '/extensions/wikia/Flags/specials/templates/SpecialFlags_createFlagForm.mustache,/extensions/wikia/Flags/specials/templates/createFormParam.mustache',
+						styles: '/extensions/wikia/Flags/specials/styles/CreateForm.scss'
+					}
+				});
+
+				cache.set(getResourcesCacheKey(), formResources, cache.CACHE_LONG);
+			}
+
+			return formResources;
+		}
+
+		function setupForm(formResources) {
+			loader.processStyle(formResources.styles);
+			mw.messages.set(formResources.messages);
+
+			formData = {
+				/* TODO - Do something with the damn messages */
+				messages: formResources.messages,
+				template: formResources.mustache[0],
+				partialParam: formResources.mustache[1]
+			};
+
+			modalConfig = {
+				vars: {
+					id: 'FlagEditForm',
+					classes: ['flag-edit-form'],
+					size: 'medium', // size of the modal
+					buttons: getButtons(),
+					content: '' // content
+				}
+			};
+		}
+
+		function getButtons() {
+			return [
+				{
+					vars: {
```

Comment:
I think these buttons miss i18n, right?
It can be done by adding value property to vars: {}
eg. value: mw.message('msg-name').escaped(),